File: php/src/Model/ParallelBranch.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * One branch of a parallel step.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class ParallelBranch
{
    public function __construct(
        public readonly string $stepId,
        public readonly string $assignmentId,
        public readonly string $status,
        /** @var array<string, mixed>|null */
        public readonly ?array $result,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            stepId: (string) ($data['stepId'] ?? ''),
            assignmentId: (string) ($data['assignmentId'] ?? ''),
            status: (string) ($data['status'] ?? ''),
            result: $data['result'] ?? null,
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/ParallelBranch.java
package io.fivexer.sdk.model;

import java.util.Map;

/**
 * One branch of a parallel step.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class ParallelBranch {
    private String stepId;
    private String assignmentId;
    private String status;
    private Map<String, Object> result;

    public String getStepId() { return stepId; }
    public String getAssignmentId() { return assignmentId; }
    public String getStatus() { return status; }
    public Map<String, Object> getResult() { return result; }
}
